lib/canvas/canvas.ts

```diff
@@ -294,7 +294,9 @@ const getFileName = (
     return path;
   }
   const limit = 50;
-  const [name, extention] = path.split('.');
+  const idx = path.lastIndexOf('.');
+  const name = path.substring(0, idx);
+  const extention = path.substring(idx);
```

Comment:
```js
const extention = path.substring(idx + 1);
```
THis should be plus 1 as it includes the `.` so `bob.canvas` becomes `bob-0..canvas`

Reply:
good catch, somehow i missed that ;)